Realm/RLMRealm.mm

```diff
@@ -766,6 +766,32 @@ + (NSUInteger)schemaVersionAtPath:(NSString *)realmPath encryptionKey:(NSData *)
     return RLMRealmSchemaVersion(realm);
 }
 
++ (BOOL)migrationRequiredAtPath:(NSString *)realmPath encryptionKey:(NSData *)key error:(NSError **)outError {
+    RLMRealm *realm = RLMGetThreadLocalCachedRealmForPath(realmPath);
+    if (realm) {
+        // we've already successfully opened a realm, so a migration isn't necessary.
+        return NO;
+    } else {
+        NSError *error;
+        key = validatedKey(key) ?: keyForPath(realmPath);
+        realm = [[RLMRealm alloc] initWithPath:realmPath key:key readOnly:YES inMemory:NO dynamic:YES error:&error];
+        if (error) {
+            RLMSetErrorOrThrow(error, outError);
```

Comment:
Ignore anything which wouldn't result in an error when actually opening the Realm (which I think if just file not found, but I'm not certain).